src/loader.js

```diff
@@ -146,15 +147,99 @@
   };
 
   // 4. Registry Objects
+  // For now, registry objects are a work in progress that don't fully integrate into the rest of the code base
 
   // 4.1.1
+  function createRegistry() {
+    var registry = new RegistryPrototype();
+    registry._registry = {};
```

Comment:
This corresponds to https://whatwg.github.io/loader/#registry-CreateRegistry. However, depending on whether `Registry` is public or not, we could potentially take some shortcuts that deviate from the spec but make the code smaller/simpler